NuRadioMC/simulation/simulation.py

```diff
@@ -426,9 +438,15 @@ def run(self):
                             temp_reflection_case = sg_pre['ray_tracing_reflection_case'][self._iE][channel_id]
                         r.set_solution(sg_pre['ray_tracing_C0'][self._iE][channel_id], sg_pre['ray_tracing_C1'][self._iE][channel_id],
                                        sg_pre['ray_tracing_solution_type'][self._iE][channel_id], temp_reflection, temp_reflection_case)
-                    else:
+                    elif find_solutions:
```

Comment:
the logic is implemented in a non ideal way (at least to me): Why don't you just add a `continue` to line 431? and also the debug output? 

Then you don't need to touch the rest of the file

Reply:
No reason except that I found it easier to read like this, and I find that the simulation module is already too dense, so a bit of code for dummies shouldn't be too bad. I can change it if you like.

Reply:
yes please change it. This cut should just be a single if block